iceoryx2-bb/cxx/include/iox2/container/static_string.hpp

```diff
@@ -477,18 +539,132 @@ class StaticString {
         return ret;
     }
 
+    /// Creates a substring containing the characters from pos until count; if pos+count is greater than the
+    /// size of the original string the returned substring only contains the characters from pos until size().
+    ///
+    /// @return an Optional containing the substring
+    ///         nullopt if pos is greater than the size of the original string
```

Comment:
This can also fail with nullopt if the resulting string would not be a valid UTF-8 encoded string (which can never happen with our current restriction of disallowing all code points above U+0080, but the entire API is designed to be forward-compatible in that way).